python/pyspark/sql/readwriter.py

```diff
@@ -593,7 +593,7 @@ def mode(self, saveMode):
 
         * `append`: Append contents of this :class:`DataFrame` to existing data.
         * `overwrite`: Overwrite existing data.
-        * `error` or `errorifexists`: Throw an exception if data already exists.
+        * `error`, `errorifexists`, `default`: Throw an exception if data already exists.
```

Comment:
It seems that `PySpark` and `SparkR` API hides this intentionally until now.

@HyukjinKwon . Do we need to update here?
@felixcheung . Do we need to add `default` mode description to `DataFrame.R`, too?

Reply:
In an opposite direction, we can hide `default` mode for all API docs if we are not encourage this.

Reply:
Interesting, I had also not heard of a "default" option. I wonder what the use case is, as you would set this at the time of creation anyway. I think it's fine to change the behavior, below, to truly make it the 'default'. If in doubt, let's not mention it in docs.

Reply:
> I wonder what the use case is, ...

Let's say you develop a library with a function which takes a `Dataframe` and some options including the write mode from the caller, and writes this data. The caller or user can set the write mode or not. So, instead of introducing additional if-else in your code, you can set the default value. Especially, it is convenient when there are many chained method calls.

Reply:
I see, that's a reasonable use case, though minor. I wonder if we support 'default' consistently everywhere though? I'd just be reluctant to advertise it yet if not.

Reply:
Thank you. I updated this PR by removing those doc changes.